src/button/internal.tsx

```diff
@@ -70,10 +73,14 @@ export const InternalButton = React.forwardRef(
           fireCancelableEvent(onFollow, { href, target }, event);
         }
 
+        if (funnelInteractionId && variant === 'primary') {
```

Comment:
Just a quick note here for offline discussion. This is a good example of an assumption about the relationship between the funnel context, the button variant, and the correct/incorrect implementation of a pattern that could be incorrect.

Reply:
As discussed in DM, there are a few options here to improve this. For now we'll continue as is and iterate on received data.